flink-autoscaler/src/main/java/org/apache/flink/autoscaler/metrics/ScalingMetric.java

```diff
@@ -65,15 +69,35 @@ public enum ScalingMetric {
     SCALE_DOWN_RATE_THRESHOLD(false),
 
     /** Expected true processing rate after scale up. */
-    EXPECTED_PROCESSING_RATE(false);
+    EXPECTED_PROCESSING_RATE(false),
 
-    private final boolean calculateAverage;
+    /**
+     * Maximum GC pressure across taskmanagers. Percentage of time spent garbage collecting between
+     * 0 (no time in GC) and 1 (100% time in GC).
+     */
+    GC_PRESSURE(false),
+
+    /** Percentage of max heap used (between 0 and 1). */
+    HEAP_USAGE(true);
+
+    @Getter private final boolean calculateAverage;
+
+    /** List of {@link ScalingMetric}s to be reported as per vertex Flink metrics. */
+    public static final List<ScalingMetric> REPORTED_VERTEX_METRICS =
+            List.of(
+                    LOAD,
+                    TRUE_PROCESSING_RATE,
+                    TARGET_DATA_RATE,
+                    CATCH_UP_DATA_RATE,
+                    LAG,
+                    PARALLELISM,
+                    RECOMMENDED_PARALLELISM,
+                    MAX_PARALLELISM,
+                    SCALE_UP_RATE_THRESHOLD,
+                    SCALE_DOWN_RATE_THRESHOLD,
+                    EXPECTED_PROCESSING_RATE);
```

Comment:
nit: Could we add one filed in this enum to indicate should each enum report vertex metrics?

Add a list can work, but other developers or we will forget it when adding one new enum in the future. If we add one field, every one cannot forget it.

Reply:
> I have added a test to compare the list of evaluated metrics to this to help developers when adding new metrics.

Sorry, could you please clarify which test will fail when one new metric should be added to `REPORTED_VERTEX_METRICS` set but developers forget it? I didn't find it.